__tests__/integration/hathorwallet_facade.test.js

```diff
@@ -0,0 +1,1665 @@
+      const [htrBalance] = await hWallet.getBalance(HATHOR_TOKEN_CONFIG.uid);
+      return htrBalance.balance.unlocked;
+    }
+
+    // Setting up scenario
+    const hWallet = await generateWalletHelper();
+    await GenesisWalletHelper.injectFunds(hWallet.getAddressAtIndex(0), 20);
+    const { hash: tokenUid } = await createTokenHelper(
+      hWallet,
+      'Token to Melt',
+      'TMELT',
+      1900
+    );
+    let expectedHtrFunds = 1;
+
+    let meltResponse;
+    // Melting less than 100 tokens recovers 0 HTR
+    meltResponse = await hWallet.meltTokens(tokenUid, 99);
+    await waitForTxReceived(hWallet, meltResponse.hash);
+    expect(await getHtrBalance(hWallet)).toBe(expectedHtrFunds);
+
+    // Melting exactly 100 tokens recovers 1 HTR
+    await waitUntilNextTimestamp(hWallet, meltResponse.hash);
+    meltResponse = await hWallet.meltTokens(tokenUid, 100);
+    expectedHtrFunds += 1;
+    await waitForTxReceived(hWallet, meltResponse.hash);
+    expect(await getHtrBalance(hWallet)).toBe(expectedHtrFunds);
+
+    // Melting less than 200 tokens recovers 1 HTR
+    await waitUntilNextTimestamp(hWallet, meltResponse.hash);
+    meltResponse = await hWallet.meltTokens(tokenUid, 199);
+    expectedHtrFunds += 1;
+    await waitForTxReceived(hWallet, meltResponse.hash);
+    expect(await getHtrBalance(hWallet)).toBe(expectedHtrFunds);
+
+    // Melting exactly 200 tokens recovers 2 HTR
+    await waitUntilNextTimestamp(hWallet, meltResponse.hash);
+    meltResponse = await hWallet.meltTokens(tokenUid, 200);
+    expectedHtrFunds += 2;
+    await waitForTxReceived(hWallet, meltResponse.hash);
+    expect(await getHtrBalance(hWallet)).toBe(expectedHtrFunds);
+
+    // Melting less than 300 tokens recovers 2 HTR
+    await waitUntilNextTimestamp(hWallet, meltResponse.hash);
+    meltResponse = await hWallet.meltTokens(tokenUid, 299);
+    expectedHtrFunds += 2;
+    await waitForTxReceived(hWallet, meltResponse.hash);
+    expect(await getHtrBalance(hWallet)).toBe(expectedHtrFunds);
+  });
+});
+
+describe('delegateAuthority', () => {
+  afterEach(async () => {
+    await stopAllWallets();
+    await GenesisWalletHelper.clearListeners();
+  });
+
+  it('should delegate authority between wallets', async () => {
+    const hWallet1 = await generateWalletHelper();
+    const hWallet2 = await generateWalletHelper();
+
+    // Creating a Custom Token on wallet 1
+    await GenesisWalletHelper.injectFunds(hWallet1.getAddressAtIndex(0), 10);
+    const { hash: tokenUid } = await createTokenHelper(
+      hWallet1,
+      'Delegate Token',
+      'DTK',
+      100,
+    );
+
+    // Delegating mint authority to wallet 2
+    const { hash: delegateMintTxId } = await hWallet1.delegateAuthority(
+      tokenUid,
+      'mint',
+      hWallet2.getAddressAtIndex(0)
+    );
+    await waitForTxReceived(hWallet1, delegateMintTxId);
+
+    /*
+     * Authority Token delegation usually takes longer than usual to be reflected on the local
```

Comment:
That's also weird. We should create an issue to investigate that.

Reply:
Opened this issue to investigate:
- #399